types/instance.go

```diff
@@ -70,3 +80,75 @@ func (i Instance) GetItem() interface{} {
 func (i Instance) GetType() string {
 	return "instance"
 }
+
+type JsonResponseBody struct {
+	Json map[string]interface{}
+}
+
+type RemoteResponse struct {
+	Code  int
+	Body  string
+	Json  JsonResponseBody
+	Error bool
+}
+
+// GetUrl returns the result of an HTTP request to the instance
+func (i Instance) GetUrl(path string, port string) RemoteResponse {
+	var uri string
+	checkPort := "80"
+
+	if port == "" {
+		uri = fmt.Sprintf("http://%s%s", i.IpAddress, path)
+	} else {
+		uri = fmt.Sprintf("http://%s:%s%s", i.IpAddress, port, path)
+		checkPort = port
+	}
+
+	if tcp_port_test(i.IpAddress, checkPort) == false {
+		return RemoteResponse{999, "", JsonResponseBody{}, true}
+	}
+
+	log.Debugf("Making HTTP request to %s", uri)
+	resp, err := http.Get(uri)
+	if err != nil {
+		log.Errorf("Error fetching %s - %s", uri, err)
+		return RemoteResponse{999, "", JsonResponseBody{}, true}
+	}
+	log.Debug("HTTP response: ", resp.Status)
+
+	if resp.ContentLength < 1 {
+		log.Debug("HTTP request returned empty response.")
+		return RemoteResponse{resp.StatusCode, "", JsonResponseBody{}, true}
+	}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.Error("Error reading response body", err)
+	}
+
+	if resp.Header.Get("Content-Type") == "application/json" {
+		jsonbody := JsonResponseBody{}
+		json.Unmarshal(body, &jsonbody)
+		log.Debug("HTTP Body: ", jsonbody)
+		return RemoteResponse{resp.StatusCode, "", jsonbody, false}
+	}
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(resp.Body)
+	return RemoteResponse{resp.StatusCode, buf.String(), JsonResponseBody{}, false}
+}
+
+func tcp_port_test(host string, port string) bool {
```

Comment:
This function name doesn't follow naming convention of the project. Could you please eliminate low dashes?